### PR TITLE
Using libuv to manager thread pool

### PR DESCRIPTION
**Pull request by luohaha**
libuv provides a thread pool, and I use it here to manager download and process threads. The thread pool's size can be initialized by `setenv(UV_THREADPOOL_SIZE, "", 1);` . Before this, I have misunderstanded libuv's thread pool.

### CODE REVIEW

File: core/scheduler/watcher.c [hunk 1]
```diff
@@ -5,16 +5,14 @@
   @handle : the uv_idle_t
 **/
 void watcher(uv_prepare_t *handle) {
+  printf("test !\n");
```
Comment (AntonioCS) [inline core/scheduler/watcher.c]: You forgo to remove this printf

Comment (luohaha) [inline core/scheduler/watcher.c]: You are right! Thanks!

**Comment by luohaha** [comment]
Test code is removed!

**Comment by AntonioCS** [comment]
:+1:

**Pull request merged**